src/runtime_src/core/tools/common/tests/TestNPU3Latency.cpp

```diff
@@ -0,0 +1,152 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2024 Advanced Micro Devices, Inc. All rights reserved.
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+// Local - Include Files
+#include "TestNPU3Latency.h"
+#include "tools/common/XBUtilities.h"
+#include "xrt/xrt_bo.h"
+#include "xrt/xrt_device.h"
+#include "xrt/xrt_hw_context.h"
+#include "xrt/xrt_kernel.h"
+#include "experimental/xrt_elf.h"
+#include "experimental/xrt_ext.h"
+#include "experimental/xrt_module.h"
+namespace XBU = XBUtilities;
+
+#include <filesystem>
+
+static constexpr size_t host_app = 1; //opcode
+static constexpr size_t buffer_size = 20;
+static constexpr int itr_count = 10000;
+
+// ----- C L A S S   M E T H O D S -------------------------------------------
+TestNPU3Latency::TestNPU3Latency()
+  : TestRunner("latency", "Run end-to-end latency test")
+{}
+
+boost::property_tree::ptree
+TestNPU3Latency::run(std::shared_ptr<xrt_core::device> dev)
+{
+  boost::property_tree::ptree ptree = get_test_header();
+  ptree.erase("xclbin");
+  ptree.erase("elf");
+
+  const auto xclbin_name = xrt_core::device_query<xrt_core::query::xclbin_name>(dev, xrt_core::query::xclbin_name::type::nop);
+  auto xclbin_path = findPlatformFile(xclbin_name, ptree);
+  if (!std::filesystem::exists(xclbin_path))
+    return ptree;
+
+  logger(ptree, "Xclbin", xclbin_path);
+
+  xrt::xclbin xclbin;
+  try {
+    xclbin = xrt::xclbin(xclbin_path);
+  }
+  catch (const std::runtime_error& ex) {
+    logger(ptree, "Error", ex.what());
+    ptree.put("status", test_token_failed);
+    return ptree;
+  }
+
+  //Elf flow
+  const auto elf_name = xrt_core::device_query<xrt_core::query::sequence_name>(dev, xrt_core::query::sequence_name::type::nop);
+  auto elf_path = findPlatformFile(elf_name, ptree);
+  if (!std::filesystem::exists(elf_path))
+    return ptree;
+
+  logger(ptree, "Elf", elf_path);
+
+  xrt::elf elf;
+  try {
+    elf = xrt::elf(elf_path);
```

Comment:
we need elf for noop? We didn't need dpu seq on prev devices so wondering what has changed.